### PR TITLE
[couchbase] Add Integration Package with CBL Replication Data Stream

### PR DESCRIPTION
**Pull request by kush-elastic**
- Enhancement

## What does this PR do?

- Generated the skeleton of Couchbase integration package.
- Added 1 data stream ( cbl_replication )
- Added data collection logic.
- Added the ingest pipelines.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added system test cases.
Note : This PR won't include dashboards. The dashboards for sync gateway will be included in the upcoming PRs. 
## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's changelog.yml file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's manifest.yml file to point to the latest Elastic stack release (e.g. ^7.13.0).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/couchbase directory.
- Run the following command to run tests. 
> elastic-package test 

ss1-cbl
ss2-cbl

## Related issues

- Relates #3666
- Relates #3706
- Relates #3771
- Relates #3900

### CODE REVIEW

**Review by yug-elastic — APPROVED** [review]

File: packages/couchbase/data_stream/cbl_replication/agent/stream/stream.yml.hbs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+metricsets: ["collector"]
+hosts:
+{{#each hosts}}
+  - {{this}}
+{{/each}}
+period: {{period}}
+metrics_path: /metrics
+metrics_filters:
```
Comment (ManojS-shetty) [inline packages/couchbase/data_stream/cbl_replication/agent/stream/stream.yml.hbs:8]: why we are using a metrics_filters here?

Comment (kush-elastic) [reply packages/couchbase/data_stream/cbl_replication/agent/stream/stream.yml.hbs:8]: We have used this [filter](https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-metricset-prometheus-collector.html#_filtering_metrics_2) option available to capture the CBL replication specific metrics from the entire Prometheus response.

**Review by ManojS-shetty — APPROVED** [review]

**Pull request merged**